torch/quantization/fx/_equalize.py

```diff
@@ -0,0 +1,221 @@
+        if not (w_orig.ndim == 2):
+            raise ValueError("WeightEqualObserver only supports Linear layers")
+
+        return self._forward(w_orig)
+
+    def _forward(self, w_orig):
+        r"""
+        Calculates the min/max values of each weight column and weight row.
+        """
+
+        w_orig = self.weight_col_obs(w_orig)
+        w_orig = self.weight_row_obs(w_orig)
+
+        # Calculate the column indices of the min/max weight in each row
+        num_row, _ = w_orig.shape
+        min_weights_ind = []
+        max_weights_ind = []
+        for i in range(num_row):
+            min_weights_ind.append(torch.nonzero(w_orig[i] == self.weight_row_obs.min_vals[i])[0][0])
+            max_weights_ind.append(torch.nonzero(w_orig[i] == self.weight_row_obs.max_vals[i])[0][0])
+        self.min_weights_ind = torch.tensor(min_weights_ind)
+        self.max_weights_ind = torch.tensor(max_weights_ind)
+
+        return w_orig
+
+    def get_weight_col_minmax(self):
+        return (self.weight_col_obs.min_vals, self.weight_col_obs.max_vals)
+
+    def get_weight_row_minmax(self):
+        return (self.weight_row_obs.min_vals, self.weight_row_obs.max_vals)
+
+    def set_equalization_scale(self, equalization_scale):
+        self.equalization_scale = equalization_scale
+
+    def calculate_qparams(self):
+        r"""
+        Returns the scale/zero_point for the input and weight rows
+        """
+
+        if self.equalization_scale.nelement() == 0:
+            warnings.warn(
+                "Must call calculate_scale before calling calculate_qparams.\
+                Returning default scale and zero point. "
+            )
+            return torch.tensor([1.0]), torch.tensor([0]), torch.tensor([1.0]), torch.tensor([0])
+
+        if self.min_weights_ind is None or self.max_weights_ind is None:
+            warnings.warn(
+                "Must find the column indicies of the minimum of each row in the \
+                weights in order to calculate the qparams calculate the \
+                qparams. Returning default scale and zero point. "
+            )
+            return torch.tensor([1.0]), torch.tensor([0]), torch.tensor([1.0]), torch.tensor([0])
+
+        # Calculate the qparams for weights by using the rows
+        # Scale the weight rows by the reciprocal of the equalization scale
+        # located at the same column index
+        (min_weights, max_weights) = self.get_weight_row_minmax()
+        min_weights_scaled = torch.mul(min_weights, torch.reciprocal(self.equalization_scale[self.min_weights_ind]))
+        max_weights_scaled = torch.mul(max_weights, torch.reciprocal(self.equalization_scale[self.max_weights_ind]))
+        (scale_weight, zero_point_weight) = self.weight_row_obs._calculate_qparams(min_weights_scaled, max_weights_scaled)
+
+        return scale_weight, zero_point_weight
+
+
+def calculate_equalization_scale(input_obs: _InputEqualObserver, weight_obs: _WeightEqualObserver) -> torch.Tensor:
```

Comment:
should we set_equalization_scale in convert instead? setting equalization scale in a `calculate_equalization_scale` doesn't sound very intuitive.

Reply:
Sure, I will move it out and instead set the equalization scale in convert().